src/app/component/CategoryProductList/CategoryProductList.component.js

```diff
@@ -22,154 +24,236 @@ import './CategoryProductList.style';
 class CategoryProductList extends Component {
     constructor(props) {
         super(props);
+
+        this.nodes = {};
+        this.observedNodes = [];
+
+        this.loadPage = this.loadPage.bind(this);
+
         this.state = {
-            prevItemsLength: 0
+            pagesCount: 1
         };
     }
 
     /**
-     * Properly returng prevItemsLength even if category is switched
+     * Properly returning pagesCount even if category is switched
      * @param {*} props
      * @param {*} state
      */
     static getDerivedStateFromProps(props, state) {
-        const { items, isLoading } = props;
-        const { prevItemsLength } = state;
+        const { pages, isLoading } = props;
+        const { pagesCount } = state;
+
+        if (isLoading) return { pagesCount: 1 };
+        if (Object.keys(pages).length !== pagesCount) return { pagesCount };
```

Comment:
Why to set a pagesCount from a state, back to a state? This looks like redundant state manipulation.